### ISSUE TITLE
Test vectors for unit tests.

### ISSUE THREAD

**Issue by simsong** [mmozeiko/aes-finder#26]
I have a similar AES key finder embedded in the [bulk_extractor](https://github.com/simsong/bulk_extractor) tool. 

I am looking for a set of test data for creating unit tests to verify that this software works. 

Do you have any such test vectors?

**Comment by mmozeiko** [comment]
You could use AES validation test vectors from NIST. They are listed in this pdf: <https://csrc.nist.gov/CSRC/media/Projects/Cryptographic-Algorithm-Validation-Program/documents/aes/AESAVS.pdf>
For example all FIPS certified software verifies their AES implementation against these vectors.

**Comment by simsong** [comment]
Not really, I think, as the AES test vectors do not include the internal memory structures used by the algorithms.

**Comment by mmozeiko** [comment]
Sure, but you use aes reference implementation to expand them and dump the round keys for your actual tests.
I mean what other kind of test data would you expect? aes key is supposed to be random, so all your round keys will be pretty random data. If you don't want to use known keys, then you won't get anything better than just randomly generated keys.

**Comment by simsong** [comment]
I want test vectors for the memory structures. You are suggesting that I create my own. I can do that, but I would rather have test vectors that others have used and validated.

I guess I will create my own. Perhaps I will post them here if they are desired. This code should have a self-test..?
